enterprise_gateway/services/sessions/kernelsessionmanager.py

```diff
@@ -385,3 +387,143 @@ def _get_sessions_loc(self):
+                        self.auth = ""
+                    else:
+                        self.log.error("No such option for auth_type/EG_AUTH_TYPE")
+                else:
+                    self.log.error("Username and/or password aren't set")
+
+    def delete_sessions(self, kernel_ids):
+        """
+        Deletes kernel sessions from database
+
+        :param list of strings kernel_ids: A list of kernel ids
+        """
+        if self.enable_persistence:
+            response = requests.delete(self.webhook_url, auth=self.auth, json=kernel_ids)
+            self.log.debug(f"Webhook kernel session deleting: {kernel_ids}")
+            if response.status_code != 204:
+                self.log.error(response.raise_for_status())
+
+    def save_session(self, kernel_id):
+        """
+        Saves kernel session to database
+
+        :param string kernel_id: A kernel id
+        """
+        if self.enable_persistence:
+            if kernel_id is not None:
+                temp_session = dict()
+                temp_session[kernel_id] = self._sessions[kernel_id]
+                body = KernelSessionManager.pre_save_transformation(temp_session)
+                response = requests.post(
+                    f"{self.webhook_url}/{kernel_id}", auth=self.auth, json=body
+                )
+                self.log.debug(f"Webhook kernel session saving: {kernel_id}")
+                if response.status_code != 204:
+                    self.log.error(response.raise_for_status())
+
+    def load_sessions(self):
+        """
+        Loads kernel sessions from database
+        """
+        if self.enable_persistence:
+            response = requests.get(self.webhook_url, auth=self.auth)
+            if response.status_code == 200:
+                kernel_sessions = response.content
+                for kernel_session in kernel_sessions:
+                    self._load_session_from_file(kernel_session)
+            else:
+                self.log.error(response.raise_for_status())
+
+    def load_session(self, kernel_id):
+        """
+        Loads a kernel session from database
+
+        :param string kernel_id: A kernel id
+        """
+        if self.enable_persistence:
+            if kernel_id is not None:
+                response = requests.get(f"{self.webhook_url}/{kernel_id}", auth=self.auth)
+                if response.status_code == 200:
+                    kernel_session = response.content
+                    self._load_session_from_file(kernel_session)
+                else:
+                    self.log.error(response.raise_for_status())
+
+    def _load_session_from_response(self, kernel):
+        """
+        Loads kernel session to current session
+
+        :param dictionary kernel: Kernel session information
+        """
+        self.log.debug("Loading saved session(s)")
+        self._sessions.update(
+            KernelSessionManager.post_load_transformation(json.loads(kernel)["kernel"])
+        )
```

Comment:
Per previous comment...
```suggestion
    def _load_session_from_response(self, kernel_session: dict):
        """
        Loads kernel session to current session

        :param dictionary kernel_session: Kernel session information
        """
        self.log.debug("Loading saved session(s)")
        self._sessions.update(
            KernelSessionManager.post_load_transformation(json.loads(kernel_session)["kernel_session"])
        )
```